### DISCUSSION TITLE
Dúvida sobre singleton service locator

### DISCUSSION THREAD

**Discussion by Welingtongsouza** [balta-io/forum#308] [Tira Dúvidas] [answered]
Olá, baltianos!

Um dúvida boba sobre singleton. Caso eu tenho um objeto em memória (singleton) e tenho excessos de requisições, por exemplo, tenho 5 mil requisições **simultâneas**  e todas essas requisições utilizam a informação que está no singleton em memória, eu poderia ter problemas com isso, com algum problema de acesso ao objeto? Por exemplo, ele bloqueia o recurso em memoria enquanto está sendo usado por outro processo? Ou a memória consegue gerencia as 5 mil requisições para o mesmo objeto?

Aproveitando a enquete, por que o balta usou addTransient na DI de repositorios invés de Scoped?
services.addTransient<CustomerRepository> invés de services.addScoped<CustomerRepository>

Já que por sua vez, vai ter uma conexão aberta durante toda a requisição.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Welingtongsouza, como vai? É uma ótima pergunta :)

Eu tinha até escrito um texto aqui respondendo mas fiquei na dúvida e queria fazer um teste para ser mais assertivo na resposta.
Vou publicar algo sobre static, singleton e threads nos próximos dias, pode ser?

**Reply by Welingtongsouza** [reply]
Combinado :) fico no aguardo
